cart.mjs

```diff
@@ -12,73 +13,54 @@ import { getPercentageFromPromocode } from "./core-cart.mjs";
-    const wallet = user.wallet;
-    const name = user.firstName;
-    const surname = user.lastName;
 
-    if(products.lenght > 0){
-        console.log(`${name} ${surname} vorrebbe comprare `, products);
-    }
-    else if(products.lenght = 0){
-        console.log(`${name} ${surname} ha il carrello vuoto`);
-        }
-    if(wallet > 0){
-        console.log(`${name} ${surname} ha ${wallet} euro`);
-    }
-    //products.forEach(product => {
+    console.log(`${separatore('*',55)}`);
     for (let currentProduct of products) {
-        //console.log('Voglio estrarre', currentProduct);
         let prodotto = core.getProduct(currentProduct);
-        console.log(`\t[${prodotto.ean}]\t${prodotto.name}\t\t${prodotto.price}`)
+        console.log(`  [${prodotto.ean}]\t${core.formatProductName(prodotto.name)}\t\t${prodotto.price} euro`)
         totOrdine += prodotto.price;
-        //console.log('Prodotto estratto con nome:', prodotto.price);
+    }
+    
+    let discountedPriceValue = core.discountedPrice(totOrdine, rate);
+    if(totOrdine !== 0){
+    console.log(`${separatore('*',55)}`);
+    console.log(`  TOTALE:\t\t\t${(totOrdine).toFixed(2)} euro`);
+    console.log(`${separatore('+',55)}`);
+    }else{
+        console.log(`${nomeUtente} non ha prodotti nel carrello`);
     }
 
     if(promoUtente !== ''
         && promoUtente !== undefined
         && promoUtente !== null){
-            console.log(`\t CODICE PROMO: \t\t ${promoUtente}`)
-            let discountedPriceValue = core.discountedPrice(totOrdine, rate);
-            console.log(`PRODOTTO SCONTATO\t\t`, discountedPriceValue)
+            console.log(`  CODICE PROMO: \t\t ${promoUtente}`)
+          
+            console.log(`  SCONTO:\t\t\t  ${(totOrdine*rate).toFixed(2)} euro`);
+            console.log(`  TOTALE SCONTATO:\t\t${discountedPriceValue} euro`);
+            console.log(`${separatore('+',55)}`);
         }
-
-    if(wallet < discountedPrice){
-        console.log(`${discounteredPrice(totOrdine,getPercentageFromPromocode(promoCode.name))}`)
-    }
+    let creditoResiduo = core.creditoResiduo(wallet, discountedPriceValue);
+    if(creditoResiduo > 0){
     
-    console.log('\t\tFINE DEL NODO\n\n')
-    //})
-
-    // console.log('RIGA DEL CARRELLO DA STAMPARE', cartRow)
-    // console.log('UUID', uuid)
-    // console.log('Prodotti', products)
-    // console.log('User completo', user);
-
-    //trovare uno user dato il carrello
-
-
-    // capire se l'utente può comprare
-
-    // capire se l'utente ha delle promo
-
-    //...
-
-    // genera ricevuta
+    console.log(`\n${separatore('**',55)}`);
+    console.log(`  ${nomeUtente} ha un credito residuo di ${creditoResiduo.toFixed(2)} euro`);
+    console.log(`${separatore('**',55)}`);
+    }else{
+        console.log(`  ${nomeUtente} NON HA ABBASTANZA SOLDI PER ACQUISTARE.`);
+        console.log(`${separatore('**',55)}`);
+    }
+    console.log('[FINE DEL NODO]\n\n\n');
```

Comment:
se ti senti di generare anche il file txt c'è poco da modificare. Prova a salvare tutta la ricevuta in una variabile `content` e poi fare `fs.writeFileSync(filename, content)`

filename deve essere qualcosa del genere: `receipts/${cartUserUUID}_receipt_${datacorrente}.txt`